charts/zot/templates/deployment.yaml

```diff
@@ -35,16 +35,50 @@ spec:
             - name: zot
               containerPort: 5000
               protocol: TCP
+          {{- if or .Values.mountConfig .Values.persistence }}
+          volumeMounts:
+          {{- if .Values.mountConfig }}
+            - mountPath: '/etc/zot'
+              name: {{ .Release.Name }}-secret
+          {{- end }}
+          {{- if and .Values.persistence }}
+            - mountPath: '/var/lib/registry'
+              name: {{ .Release.Name }}-volume
+          {{- end }}
+          {{- end }}
           livenessProbe:
             httpGet:
               path: /v2/
               port: 5000
+              {{- if .Values.authHeader }}
+              httpHeaders:
+                - name: Authorization
+                  value: Basic {{ .Values.authHeader }}
+              {{- end }}
           readinessProbe:
             httpGet:
               path: /v2/
               port: 5000
+              {{- if .Values.authHeader }}
+              httpHeaders:
+                - name: Authorization
+                  value: Basic {{ .Values.authHeader }}
+              {{- end }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+      {{- if or .Values.mountConfig .Values.persistence }}
+      volumes:
+      {{- if .Values.mountConfig }}
+        - name: {{ .Release.Name }}-secret
+          secret:
+            secretName: {{ .Release.Name }}-secret
```

Comment:
Actually, I prefer to use configmap for config file. Any reason to use secret?

Reply:
Yes, I put passwords on the config, so ...

Reply:
I reviewed my usage and in fact the passwords are on a separate `htpasswd` file, so probably a `configMap` and a Secret could be used instead of a single secret... as it was for personal use I guess I did what was faster at the time.

If you are interested in the patch I can add the support for a secret and a configMap.

Reply:
Also done on my branch